server/app/services/UserService.ts

```diff
@@ -64,7 +71,55 @@ export class UserService {
     if (!updatedUser) {
       throw new NotFoundError(`User with username ${username} was not found`);
     }
-
     return UserEntity.buildUser(updatedUser);
   }
+
+  public async forgotPassword(url: string, body: IUser): Promise<string> {
+    const email = body.email;
+    const user: IUserEntity = UserEntity.buildUser(await this.userRepo.findByEmail(body));
+    const accessToken = generateToken(user);
+    requestMap.set(email, accessToken);
+
+    const info: SentMessageInfo = await this.systemService.sendEmail({
+      to: [email],
+      subject: 'ERP Password Reset',
+      emailBody: `This is the ERP Team07,\n\nPlease reset your password using the following link: ${url}/reset/${accessToken}\n\nRegards.`,
+    });
+    return info;
+  }
+
+  public async resetPassword(token: string, pass: string) {
+    let email = '';
+    let found = false;
+
+    try {
+      jwt.verify(token, config.get<string>('jwt.secret'), function (
```

Comment:
Since this is copy pasted from the middleware file, can you just extract this into a method so we can reuse it in both places

Reply:
no stress if youre unable

Reply:
@Mackbellemore I guess I'll leave this as it is, I need to retrieve value from the callback in the first place